src/ssl_certman.c

```diff
@@ -819,7 +819,11 @@ int wolfSSL_CertManagerVerify(WOLFSSL_CERT_MANAGER* cm, const char* fname,
         WOLFSSL_MSG("Getting dynamic buffer");
         buff = (byte*)XMALLOC((size_t)sz, cm->heap, DYNAMIC_TYPE_FILE);
         if (buff == NULL) {
-            ret = WOLFSSL_BAD_FILE;
+            /* If some other error has already occurred preserve that code
+             * return WOLFSSL_BAD_FILE if that is the only error condition */
+            if (ret == WOLFSSL_SUCCESS) {
```

Comment:
Hang on. Why are we not enabling the check of ret above instead of doing the malloc when we are already failing.

Reply:
Great question! I did start by doing the check above but the compiler complained a code path existed such that `buff` was uninitialized:

```
make[1]: Entering directory '/home/kaleb/fork-wolfssl'
make[2]: Entering directory '/home/kaleb/fork-wolfssl'
make[2]: warning: -j5 forced in submake: resetting jobserver mode.
  CC       src/libwolfssl_la-ssl.lo
In file included from ./wolfssl/internal.h:27,
                 from src/ssl.c:37:
./src/ssl_certman.c: In function ‘wolfSSL_CertManagerVerify’:
./wolfssl/wolfcrypt/types.h:573:70: error: ‘buff’ may be used uninitialized [-Werror=maybe-uninitialized]
  573 |                     #define XFREE(p, h, t)       {void* xp = (p); if (xp) wolfSSL_Free(xp);}
      |                                                                      ^
In file included from src/ssl.c:213:
./src/ssl_certman.c:785:12: note: ‘buff’ was declared here
  785 |     byte*  buff;
      |            ^~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:7041: src/libwolfssl_la-ssl.lo] Error 1
make[2]: Leaving directory '/home/kaleb/fork-wolfssl'
make[1]: *** [Makefile:7825: all-recursive] Error 1
make[1]: Leaving directory '/home/kaleb/fork-wolfssl'
make: *** [Makefile:4663: all] Error 2
```

This was my initial patch that produced the above error:

```
diff --git a/src/ssl_certman.c b/src/ssl_certman.c
index 9f227be15..bb6756182 100644
--- a/src/ssl_certman.c
+++ b/src/ssl_certman.c
@@ -814,16 +814,15 @@ int wolfSSL_CertManagerVerify(WOLFSSL_CERT_MANAGER* cm, const char* fname,
      * small. */
 #ifndef WOLFSSL_SMALL_STACK
     if ((ret == WOLFSSL_SUCCESS) && (sz > (long)sizeof(staticBuffer)))
+#else
+
+    if (ret == WOLFSSL_SUCCESS)
 #endif
     {
         WOLFSSL_MSG("Getting dynamic buffer");
         buff = (byte*)XMALLOC((size_t)sz, cm->heap, DYNAMIC_TYPE_FILE);
         if (buff == NULL) {
-            /* If some other error has already occurred preserve that code
-             * return WOLFSSL_BAD_FILE if that is the only error condition */
-            if (ret == WOLFSSL_SUCCESS) {
-                ret = WOLFSSL_BAD_FILE;
-            }
+            ret = WOLFSSL_BAD_FILE;
         }
     }
     /* Read all the file into buffer. */
```

Reply:
We can go with that approach instead and do the initialization to satisfy the compiler:

```
diff --git a/src/ssl_certman.c b/src/ssl_certman.c
index 9f227be15..46011595f 100644
--- a/src/ssl_certman.c
+++ b/src/ssl_certman.c
@@ -782,7 +782,7 @@ int wolfSSL_CertManagerVerify(WOLFSSL_CERT_MANAGER* cm, const char* fname,
 #ifndef WOLFSSL_SMALL_STACK
     byte   staticBuffer[FILE_BUFFER_SIZE];
 #endif
-    byte*  buff;
+    byte*  buff = NULL;
     long   sz = 0;
     XFILE  file = XBADFILE;
 
@@ -814,16 +814,15 @@ int wolfSSL_CertManagerVerify(WOLFSSL_CERT_MANAGER* cm, const char* fname,
      * small. */
 #ifndef WOLFSSL_SMALL_STACK
     if ((ret == WOLFSSL_SUCCESS) && (sz > (long)sizeof(staticBuffer)))
+#else
+
+    if (ret == WOLFSSL_SUCCESS)
 #endif
     {
         WOLFSSL_MSG("Getting dynamic buffer");
         buff = (byte*)XMALLOC((size_t)sz, cm->heap, DYNAMIC_TYPE_FILE);
         if (buff == NULL) {
-            /* If some other error has already occurred preserve that code
-             * return WOLFSSL_BAD_FILE if that is the only error condition */
-            if (ret == WOLFSSL_SUCCESS) {
-                ret = WOLFSSL_BAD_FILE;
-            }
+            ret = WOLFSSL_BAD_FILE;
         }
     }
     /* Read all the file into buffer. */
```

Which is your preference?

Reply:
Fixed.